### ISSUE TITLE
Model Deployment

### ISSUE THREAD

**Issue by rohithsiddhartha** [google-research/pegasus#84]
Did anyone try to deploy pegasus model either from checkpoints created or after creation of savedmodel.pb and variables.

I tried it out it in AI-platform to deploy but there was a size limit of 2GB. Experimented with quantization, creation of frozen model and then 
I incurred with the similar error
Traceback (most recent call last):
  File "freezemodel.py", line 81, in <module>
    freeze_graph(model_dir)
  File "freezemodel.py", line 75, in freeze_graph
    f.write(output_graph_def.SerializeToString())
ValueError: Message tensorflow.GraphDef exceeds maximum protobuf size of 2GB: 2296684200. 

Can anyone help me out in providing a solution for deployment or anyone who has worked on deployment of large models >2gb especially for online prediction which runs with a minimal latency.

**Comment by abodacs** [comment]
Hi @rohithsiddhartha 
Could you check/try huggingface implementation?

https://huggingface.co/models?filter=pegasus

**Comment by rohithsiddhartha** [comment]
Hi @abodacs 
You have provided a nice UI for prediction but the latency is way higher than it should have been.

I tried to deploy model in docker using Flask and ran inference by this I was able to save model loading time as it is up and running all the time in the docker once it is ran. 
Inference time came up to around 10 sec on an average, even though it isn't good enough to do online predictions, later part of deployment work using kubernetes is still under progress. 

I was able to finish my comment on the issue and still it didn't summarize yet, may be there might be some issue in the back end or let me know if the current latency is normal. 

I'm closing this issue.
